WcaOnRails/app/models/competition.rb

```diff
@@ -21,12 +21,15 @@ class Competition < ActiveRecord::Base
   MAX_ID_LENGTH = 32
   MAX_NAME_LENGTH = 50
   validates :id, presence: true, uniqueness: true, length: { maximum: MAX_ID_LENGTH },
-                 format: { with: /\A[a-zA-Z0-9]+\Z/ }
+                 format: { with: /\A[a-zA-Z0-9]+\Z/ }, if: :name_valid_or_updating?
+  private def name_valid_or_updating?
+    self.persisted? || name.length <= MAX_NAME_LENGTH && name =~ ENDS_WITH_YEAR_RE
```

Comment:
So, I'm still confused. Based on the boolean logic, this looks more like `name_valid_or_creating?`, or even `creating_or_name_valid?`.

> I want to perform this validation all the time on the edit page. Because someone could update the name and the cellName on the validation page and at that point I want to display both errors to the user. It's only on the new page that I don't want to validate the id and cellName if the name itself isn't valid.

I see. I'm nervous that this validation would allow people to create new competitions with invalid `id`s and `cellName`s, though. You could achieve this by adding a form manually via javascript, or using curl.

For now, I think it would be okay to just always enforce these validations. Admittedly, it would be confusing to users to get validation errors on fields they don't even see, but I think we could eventually port our automatic `id` and `cellName` computation to javascript, and get rid of the yucky mess inside our model to conditionally compute them. I think this would clean up our code, and provide a much nicer UX.